### PR TITLE
Don't create stdinLines just to cancel it

### PR DESCRIPTION
**Pull request by kevmoo**
Prepare to release "dot" releases of pkg:test_core and pkg:test

I plan to backport these as well...

### CODE REVIEW

File: pkgs/test_core/lib/src/util/io.dart [hunk 1]
```diff
@@ -54,9 +53,18 @@ SuitePlatform currentPlatform(Runtime runtime) => SuitePlatform(runtime,
...
-final stdinLines = StreamQueue(
+StreamQueue<String> get stdinLines => _stdinLines ??= StreamQueue(
     Platform.isFuchsia ? Stream<String>.empty() : lineSplitter.bind(stdin));
 
+StreamQueue<String>? _stdinLines;
+
+/// Call cancel on [stdinLines], but only if it's been accessed previously.
+void cancelStdinLines() {
+  if (_stdinLines != null) {
+    _stdinLines!.cancel(immediate: true);
```
Comment (natebosch) [inline pkgs/test_core/lib/src/util/io.dart:64]:
```suggestion
    _stdinLines?.cancel(immediate: true);
```

**Review by natebosch — CHANGES_REQUESTED** [review]

File: pkgs/test_core/lib/src/util/io.dart (hunk 1 above)
Comment (kevmoo) [reply pkgs/test_core/lib/src/util/io.dart:64]: I don't see how we could have a race condition here...

Comment (kevmoo) [reply pkgs/test_core/lib/src/util/io.dart:64]: ...I mean, sure. if you want. Be hyper defensive?

Comment (kevmoo) [reply pkgs/test_core/lib/src/util/io.dart:64]: Should I set it to null after then?

File: pkgs/test_core/lib/src/util/io.dart [hunk 2]
```diff
@@ -54,9 +53,18 @@ SuitePlatform currentPlatform(Runtime runtime) => SuitePlatform(runtime,
...
+  }
```
Comment (natebosch) [inline pkgs/test_core/lib/src/util/io.dart:65]:
```suggestion
  _stdinLines?.cancel(immediate: true);
```

Sorry, forgot that we can do multiline suggestions now.

Remove the conditional and only use `?.`.

**Comment by kevmoo** [comment]
LGTM @natebosch ?

**Review by natebosch — APPROVED** [review]

**Pull request merged**